\section{Conclusions}


The main goal of this project has been to develop tools and workflows to perform integrative analysis as well as their application in the analysis of public cancer data. The motivations came from the fact that integrating data from different databases, stored in different formats and with different biological meanings is a complex process from both a computational point of view by dealing with big datasets which requires an optimized use of computational resources as well as from a biological point of view for dealing with a biological process still under study, often unknown, or even without a scientific consensus.

\section{Conclusions and future studies}

In this section, we list the conclusions and future studies for the results reported in the previous chapters.

\subsection{Conclusions and future works of TCGAbiolinks}

This work has presented an effort towards the search, download and
prepare of data from the NCI's Genomic Data Commons (GDC) data portal for
downstream analysis. It offered several single-dataset
exploratory analysis such as PCA, clustering methods, box plots,
 differential gene expression analysis, differential methylated CpGs (DMCs) analysis,
 survival analysis and some integrative analysis
 such as the integration of differential methylated CpGs results with  differential gene expression analysis
 to identify regions different methylated with the nearest gene level expression changed, and
 gene set enrichment analysis.
As future works it is required the improvement of integrative analysis of the starburst plot, this method
integrates the results from two different analysis DEA (differential expression analysis) and DMR
(differential methylated regions).
There is however a problem that the results are analyzed considering the set and not the same sample,
that means the expression and methylation data of a sample are not compared to each other.
It is inferred if a data set has a differentiated average of methylation and a mean of
expression of the nearest differentiated gene, and may even be different populations.
This flaw is corrected in ELMER algorithms.

\subsection{Conclusions and future works of TCGAbiolinksGUI}

This work created a Graphical User Interface (GUI) to our command-line tools,
with the purpose to help users without programming knowledge to perform
deeper downstream analysis. Among possible improvements in future works are
the export of figures in vector formats (PDF, SVG) that do not lose quality
if altered (enlarged or reduced), the facilitation of the incorporation of external
user data, which although can be incorporated if formatted in the current defined standard,
this can still be improved, the creation of a modularization of the tool,
which would load only the packages chosen by the user in order to decrease the
amount of libraries needed to run the interface, which
would be a challenge and may not be possible due to the limitations of the development tools used R / Shiny).

\subsection{Conclusions and future works of ELMER}

This work has presented an effort towards the integrative analysis performed
using RNA-seq, DNA methylation and histone marks to identify a candidate regulatory network.
It mainly identifies distal regions with a difference of DNA methylation and correlates them
with the gene expression levels of upstream and downstream genes.
A motif enrichment analysis is performed in the regulatory regions from the anti-correlated
pairs (loss of DNA methylation and gain o gene expression) to identify potential
regulatory TF candidates. As future works we suggest to expand the algorithm to
the promoter regions, to expand the DNA methylation analysis to accept
\sigla{WGBS}{whole-genome bisulfite sequencing} data,
 and to use mutation information instead of DNA methylation
to identify regulatory regions mutated that might have affected the
regulation of a upstream/downstream genes.


\section{Publications, presentations and softwares of the Doctorate Period}

The work produced during the Doctorate period in form of scientific articles,
softwares and conference presentations is shown in the next subsections.
The published scientific articles were divided into two groups, one with the  first authorship ones, and the other with the co-authorship ones. Those are listed in the following subsection: "First-authored papers", "Co-authored papers","First-authored softwares" and "Co-authored softwares".
The subsection "workshops and workflows" contains all the material created to help users to use the developed tools. Finally, the subsection "Conferences \& presentations" list all oral and poster presentations in international and national conferences.

\subsection{First-authored papers}
\begin{itemize}
	\item \citetext{TCGAbiolinks}
	\item \citetext{10.12688/f1000research.8923.2}
	\item \citetext{Silva147496}
	\item \citetext{ELMERv2}
\end{itemize}

\subsection{Co-authored papers}

\begin{itemize}
	\item \citetext{Lingutjnl-2017-314607}
	\item \citetext{cell}
	\item \citetext{malta2017glioma}
  \item \citetext{cava2017spidermir}
  \item \citetext{foxj1}
\end{itemize}


\subsection{First-authored softwares}
\begin{description}
	\item[TCGAbiolinks (version 2.0)] An R/Bioconductor package for integrative analysis of TCGA data. Published in Bioconductor \burl{http://bioconductor.org/packages/TCGAbiolinks/}. Source code available in GitHub \burl{https://github.com/BioinformaticsFMRP/TCGAbiolinks}.
    \item[TCGAbiolinksGUI] A Graphical User Interface to analyze cancer genomics and epigenomics data. Published in GitHub \burl{https://github.com/BioinformaticsFMRP/TCGAbiolinksGUI}.
    \item[ELMER (version 2.0)] Enhancer Linking by Methylation/Expression Relationship (ELMER) is package to identify tumor-specific changes in DNA methylation within distal enhancers, and link these enhancers to downstream target genes. Published in Bioconductor \burl{http://bioconductor.org/packages/ELMER/}. Source code available in GitHub  \burl{https://github.com/tiagochst/ELMER}.
\end{description}

\subsection{Co-authored softwares}
\begin{description}
	\item[SpidermiR: An R/Bioconductor package for integrative network analysis with miRNA data]{Published in Bioconductor \burl{http://bioconductor.org/packages/SpidermiR/}. Source code available in GitHub \burl{https://github.com/claudiacava/SpidermiR}}.
\end{description}


\subsection{Workshops and workflows}
\begin{description}
\item[Workshop]{Integrative analysis workshop with TCGAbiolinks and ELMER. Dana Farber Cancer Institute, Boston, MA. Link to workshop: \burl{https://bioinformaticsfmrp.github.io/Bioc2017.TCGAbiolinks.ELMER/index.html}. Source code available in GitHub \burl{https://github.com/BioinformaticsFMRP/Bioc2017.TCGAbiolinks.ELMER}.}
\item[Workflow]{TCGA Workflow: Analyze cancer genomics and epigenomics data using Bioconductor packages. Available at \burl{https://www.bioconductor.org/help/workflows/TCGAWorkflow/}. Source code available in GitHub \burl{https://github.com/BioinformaticsFMRP/TCGAWorkflow}.}

\end{description}

\subsection{Conferences \& presentations}
\begin{description}
	\item[SNOLA 2016 - Update on Neuro-Oncology - Oral presentation - 14/19/2016]{"EPIGENOMIC AND TRANSCRIPTOMIC ANALYSIS OF ADULT GLIOMA
REVEALS CANDIDATE DRIVER TRANSCRIPTION FACTORS
INVOLVED IN GLIOMA PROGRESSION." Windsor Barra Hotel, Rio de Janeiro - Brazil}
    \item[Chromatin and Epigenetics - Poster presentation - 5/5/2017]{TCGAbiolinksGUI: A Graphical User Interface to analyze cancer genomics and epigenomics data. EMBL, Heidelberg, Germany}
	\item[Omics Seminar - Oral presentation - 06/06/2017]{Enhancer Linking by Methylation/Expression Relationship: a case study using Breast Cancer. Cedars-Sinai Medical Center, Los Angeles, California.}
	\item[Bioc2017 - Oral presentation - 07/28/2017]{Workshop: Integrative analysis workshop with TCGAbiolinks and ELMER. Dana Farber Cancer Institute, Boston, MA. Link to workshop: \burl{https://bioinformaticsfmrp.github.io/Bioc2017.TCGAbiolinks.ELMER/index.html}}
\item[From Single to Multiomics - Poster presentation - 11/13/2017]{Enhancer Linking by Methylation/Expression Relationships with the R package (ELMER) version 2. EMBL, Heidelberg, Germany}

\end{description}


%\bibliography{references}
