
\section{Glioma analysis}

The \sigla{ELMER}{Enhancer Linking by Methylation/Expression Relationship} tool was used to analyze the molecular differences between the newly identified G-CIMP-low subtype of glioma that was associated with significantly worse survival compared to the G-CIMP-high, recently described by Dr. Noushmehr and his lab \cite{ceccarelli2016molecular}. 
For this analysis, TCGA data from the NCI's Genomic Data Commons (GDC) was downloaded using our R/Bioconductor TCGAbiolinks package. Table \ref{gcimp.samples} summarizes the number of samples in each group that have both \sigla{DNA}{deoxyribonucleic acid} methylation data for the Illumina HumanMethylation450 platform (HM450) and gene expression data
(RNA-Seq). and table \ref{gcip.elmer.arg} summarizes the main values for the ELMER arguments.

% Please add the following required packages to your document preamble:
% \usepackage{booktabs}
\begin{table}[h!]
\centering
\caption[G-CIMP analysis: Sample summary]{G-CIMP-high vs G-CIMP-low analysis: number of samples with both DNA methylation (HM450) and gene expression (RNA-seq) data.}
\label{gcimp.samples}
\begin{tabular}{@{}ll@{}}
\toprule
Group       & Number of samples \\ \midrule
G-CIMP-high & 233               \\
G-CIMP-low  & 11               
\end{tabular}
\end{table}
% Please add the following required packages to your document preamble:
% \usepackage{booktabs}
\begin{table}[h!]
\centering
\caption[G-CIMP analysis: ELMER arguments values]{G-CIMP-high vs G-CIMP-low analysis: ELMER arguments values}
\label{gcip.elmer.arg}
\begin{tabular}{@{}lll@{}}
\toprule
Step                        & Argument                           & Value  \\ \midrule
createMAE           & genome &  hg38   \\
Pairs correlation/TF analysis            & Mode &  Supervised   \\
All                         & minSubgroupFrac                    & 100\%  \\
DNA methylation differences & min mean difference                & 0.3    \\
DNA methylation differences & p-value adj cut-off                & 0.01   \\
Pairs correlation           & \# permutations                    & 10000  \\
Pairs correlation           & raw p-value cut-off                & 0.001 \\
Pairs correlation           & empirical p-values cut-off         & 0.001 \\
Motif enrichment            & minimum \# probes (enriched motif) & 10     \\
Motif enrichment             & lower.OR                           & 1.1    \\ \bottomrule
\end{tabular}
\end{table}

The results are summarized in Figure \ref{tab:or}, which shows the Odds Ratio (x axis) for the enriched motifs, and in Table  \ref{tab:tf}, 
which  shows the candidate regulatory TFs whose expression anti-correlated with the DNA methylation level on the probes of each enriched motifs. From the most anti-correlated ones, ELMER uses the TFClass classification (\cite{doi:10.1093/nar/gku1064}) to identifies which TFs are  known to bind in those motifs. This classification has two levels, family and subfamily, which groups motifs with a similar signature. Depending on the motif, its family classification might have a very similar signature, otherwise, the subfamily classification is the most indicated.    

\begin{center}
\begin{figure}[h!]
\includegraphics[width=16cm]{images/hyper_motif_enrichment.pdf}
\caption[G-CIMP analysis: Odds Ratio plot]{Motif enrichment analysis: Odds Ratio (x axis) for the selected motifs with \sigla{OR}{Odds Ratio} above 1.5. The range shows the 95\% confidence interval for each Odds Ratio.}
\end{figure}
\label{tab:or}
\end{center}


For example, in Table \ref{tab:tf} the motif HXD3 has as potential TF candidate the HOXD13 if we consider the family \sigla{TF}{Transcription Factor} classification and the HOXD3 TF candidate considering the subfamily TF classification. Figure \ref{tab:hocomoco} shows the motif signature for the HOX-related factors family from \sigla{HOCOMOCO}{HOmo sapiens COmprehensive MOdel COllection} database. The transcription factors \sigla{HOXD13}{Homeobox D13} and \sigla{HOXD3}{Homeobox D3} are in the same family (HOX-related factors) but in different subfamilies.



\begin{table}[h!]
\centering
\caption[G-CIMP analysis: TF ranking plot]{TF ranking analysis: statistic For each enriched motif the anti-correlation level of all human TFs expression level with average DNA methylation level at sites with a given motif was access and ranked by the $-log_{10}(P_{value})$, the most relevant one that belongs to the same family as the motif is shown in column \textit{top.potential.TF.family} while the most relevant within the same sub-family classification is shown in column \textit{top.potential.TF.subfamily}}.
\csvautobooktabular[respect underscore]{tables/getTF.hyper.significant.TFs.with.motif.summary.csv}
\label{tab:tf}
\end{table}

\begin{center}
\begin{figure}[h!]
\includegraphics[width=16cm]{images/HOCOMOCO.png}
\caption[HOCOMOCO: HOX-related factors family]{HOCOMOCO V11: HOX-related factors family. Transcription factors HOXD13 and HOXD3 are in the same family (HOX-related factors) but in different subfamilies.}
\end{figure}
\label{tab:hocomoco}
\end{center}

To validate those findings, biological experiments are needed which by either knocking down the TF or by regulating the DNA methylation levels of those binding regions will be able to verify if the downstream genes are being regulated.


