


\section{Glioma analysis} \label{sec:glioma_analysis}

\sigla{ELMER}{Enhancer Linking by Methylation/Expression Relationship}
analysis was performed to identify the molecular differences between the newly
identified G-CIMP-low and G-CIMP-high subtypes of glioma recently described by Dr. Noushmehr
and his lab \cite{ceccarelli2016molecular}.
There is a particular interest in these molecular subtypes of glioma, since they have a great difference in both DNA methylation level and survival rate, with the G-CIMP-high subtype having a higher level of DNA methylation in some regions and a better survival rate. Thus, we seek to understand the molecular biology behind these changes leading to a worsening of survival.

For this analysis, TCGA data from the NCI's Genomic Data Commons (GDC) was
downloaded using our R/Bioconductor TCGAbiolinks package described in the previous chapter.
Table \ref{gcimp.samples} summarizes the number of samples in each group that have both \sigla{DNA}{deoxyribonucleic acid}
methylation data for the Illumina HumanMethylation450 platform (HM450) and gene expression data
(RNA-Seq) and Table \ref{gcip.elmer.arg} summarizes the main values for the ELMER arguments.
This analysis was performed with data aligned against the genome of reference hg38 and
in the "supervised" mode, which ensures that in all the steps the
comparisons are made between each group using all their samples.

% Please add the following required packages to your document preamble:
% \usepackage{booktabs}
\begin{table}[h!]
\centering
\caption[G-CIMP analysis: Sample summary]{G-CIMP-high vs G-CIMP-low analysis: number of samples with both DNA methylation (HM450) and gene expression (RNA-seq) data.}
\label{gcimp.samples}
\begin{tabular}{@{}ll@{}}
\toprule
Group       & Number of samples \\ \midrule
G-CIMP-high & 233               \\
G-CIMP-low  & 11
\end{tabular}
\end{table}
% Please add the following required packages to your document preamble:
% \usepackage{booktabs}
\begin{table}[h!]
\centering
\caption[G-CIMP analysis: ELMER arguments values]{G-CIMP-high vs G-CIMP-low analysis: ELMER arguments values.}
\label{gcip.elmer.arg}
\begin{tabular}{@{}lll@{}}
\toprule
Step                        & Argument                           & Value  \\ \midrule
createMAE           & genome &  hg38   \\
Pair probe-gene correlation/master TF             & Mode &  Supervised   \\
All                         & minSubgroupFrac                    & 100\%  \\
All                         & group1                    & GGimp-high  \\
All                         & group2                    & GGimp-low  \\
All                         & direction                    & hypermethylated probes  \\
DMCs & $\min\Delta\overline{\beta}$                & 0.3    \\
DMCs & p-value adj cut-off                & 0.01   \\
Pair probe-gene correlation           & \# permutations                    & 10000  \\
Pair probe-gene correlation           & raw p-value cut-off                & 0.001 \\
Pair probe-gene correlation           & empirical p-values cut-off         & 0.001 \\
Motif enrichment            & minimum \# probes  & 10     \\
Motif enrichment            & lower OR                           & 1.1    \\ \bottomrule
\end{tabular}
\end{table}

The DMCs (differentially methylated CpGs) analysis, which uses a one-tailed t-test to find
probes with higher mean level in G-CIMP-high group compared to the G-CIMP-low one.
The result of this analysis is summarized in Figure \ref{fig:gcimpvolcano}.
The volcano plot shows the difference of DNA methylation ($\Delta\overline{\beta}$)
versus significance $-log_{10}(\textrm{FDR corrected P-values})$. Using as default cutoffs
 $\Delta\overline{\beta}\geq0.3$ and $-log_{10}(\textrm{FDR corrected P-values})\leq 0.01$,
$7540$ probes were identified to be differentially methylated.

\begin{center}
\begin{figure}[h!]
\includegraphics[width=16cm]{images/gcimp_volcano.png}
\caption[G-CIMP analysis: Volcano plot]{\label{fig:gcimpvolcano}G-CIMP analysis: DNA methylation  Volcano plot. Difference of DNA methylation ($\min\Delta\overline{\beta}$) versus significance $-log_{10}(\textrm{FDR corrected P-values})$.}
\end{figure}

\end{center}

Those differentially methylated probes were then paired with their 10 upstream and 10 downstream genes
and those significantly anti-correlated were selected (genes have higher expression levels  when the DNA methylation levels
decreases). Using as cutoffs
$-log_{10}(\textrm{raw P-values})\leq 0.001$ and $-log_{10}(\textrm{Permuted corrected P-values})\leq 0.001$
for a  $10000$ permutation correction approach,
886 pairs of gene and probes were identified to be significantly anti-correlated.
The result of this analysis is summarized in Figure \ref{fig:gcimpheatmap}.
The heatmap is composed of other three heatmaps, one heatmap for the DNA methylation levels,
one heatmap for the gene expression levels and one with the distance between the gene and the probe.


\begin{center}
\begin{figure}[h!]

\includegraphics[width=16cm]{images/gcimp_heatmap.png}
\caption[G-CIMP analysis: Heatmap of paired probes and distal genes]{\label{fig:gcimpheatmap}G-CIMP analysis:
Heatmap of paired probes and distal genes. The first heatmap (left one) shows DNA methylation $\beta$ levels ranging from 0 (non-methylated) up to 1 (methylated probes).
The second heatmap (middle one) shows z-scores for gene expression levels (standard deviations from each gene means). The last heatmap (right heatmap) shows the distance between the probe
and gene anti-correlated. The top annotation in the heatmap was retried from \citeonline{cell}}
\end{figure}

\end{center}

\begin{table}[h!]
\centering
\caption[G-CIMP analysis: Summary results]{G-CIMP-high vs G-CIMP-low analysis:
number of distal probes, significant differentially methylated probes,
significant anti-correlated paired gene-probes and enriched motifs}
\label{gcimp.summary.results}
\begin{tabular}{@{}ll@{}}
\toprule
Result      & Value \\ \midrule
Distal probes       & 135374 \\
Significant differentially methylated probes & 7540               \\
Significant anti-correlated paired gene-probes  & 886 \\
Enriched motifs & 84
\end{tabular}
\end{table}

The  motif enrichment analysis results are summarized in Figure \ref{tab:or},
which shows the \sigla{OR}{Odds Ratio} (x axis) for the enriched motifs, and in Table  \ref{tab:tf},
which shows the candidate regulatory TFs whose expression anti-correlated
with the DNA methylation level on the probes of each enriched motifs.
 From the most anti-correlated ones, ELMER uses the TFClass
 classification \cite{doi:10.1093/nar/gku1064} to identifies
 which TFs are  known to bind in those motifs.
 This classification has two levels, family and subfamily,
  which groups motifs with a similar signature.
  Depending on the motif, its family classification might have a
  very similar signature, otherwise, the subfamily classification is the most indicated.




For example, in Table \ref{tab:tf} the motif HXD3 has as potential TF candidate the HOXD13 if we consider the family \sigla{TF}{Transcription Factor} classification and the HOXD3 TF candidate considering the subfamily TF classification. Figure \ref{tab:hocomoco} shows the motif signature for the HOX-related factors family from \sigla{HOCOMOCO}{HOmo sapiens COmprehensive MOdel COllection} database. The transcription factors \sigla{HOXD13}{Homeobox D13} and \sigla{HOXD3}{Homeobox D3} are in the same family (HOX-related factors) but in different subfamilies.



The analysis identified several candidate master regulator TF including PAX3, HMX3, FOXM1, HOXD13, HOXD11, HOXC10 and SOX11. Some of them have already been reported in previous research presented below.

%what a long sentence!  Fix it.
\citeauthor{morales2014regionalized} showed that HMX3 is a key player in the activation and maintenance of Gsh1 expression which resulted in subsequent downstream generation of GHRH (growth hormone-releasing hormone) expression in neurons \cite{morales2014regionalized}. Furthermore, GHRH antagonists in glioblastoma cells showed efficacy of those drugs for experimental treatment of glioblastoma  \cite{pozsgai2010effect}.

\citeauthor{lee2015identification} reported that HOXD13 is substantially up regulated in GBMs (Glioblastomas) and may contribute to malignancy.
Also, PAX3 and FOXM1 were shown to be essential for gliomagenesis by \citeauthor{xia2013pax3} and  \citeauthor{wang2015glioblastoma}, respectively.

However, contrary to what was expected, \citeauthor{hide2009sox11} showed that SOX11 prevents gliomagenesis, which may indicate a false positive in our analyzes.

To validate these findings, biological experiments are needed. By either knocking down these TFs or by regulating the DNA methylation levels of those binding regions will be able to verify if the downstream genes are being regulated.

\begin{center}
\begin{figure}
\includegraphics[width=16cm]{images/gcimp_motif_enrichment.pdf}
\caption[G-CIMP analysis: Odds Ratio plot]{\label{tab:or}Motif enrichment analysis: Odds Ratio (x-axis) for the selected motifs with lower OR above 1.3. The range shows the 95\% confidence interval for each Odds Ratio.}
\end{figure}
\end{center}

\begin{table}[h!]
\centering
\caption[G-CIMP analysis: TF ranking plot]{TF ranking analysis: statistic For each enriched motif the anti-correlation level of all human TFs expression level with average DNA methylation level at sites with a given motif was access and ranked by the $-log_{10}(P_{value})$, the most relevant one that belongs to the same family as the motif is shown in column \textit{top.potential.TF.family} while the most relevant within the same sub-family classification is shown in column \textit{top.potential.TF.subfamily}}.
\csvautobooktabular[respect underscore]{tables/getTF.hyper.significant.TFs.with.motif.summary.csv}
\label{tab:tf}
\end{table}

\begin{center}
\begin{figure}[h!]
\includegraphics[width=16cm]{images/HOCOMOCO.png}
\caption[HOCOMOCO: HOX-related factors family]{\label{tab:hocomoco}HOCOMOCO V11: HOX-related factors family. Transcription factors HOXD13 and HOXD3 are in the same family (HOX-related factors) but in different subfamilies.}
\end{figure}
\end{center}
