
\newcommand{\comando}[1]{\textbf{$\backslash$#1}}


\section{Objectives}

The main goal of this project is to develop tools for searching, retrieving and
analyzing pan-cancer genomic data from several databases, such as the NCI's
\sigla{GDC}{Genomic Data Commons}, which contains data from the \sigla{TCGA}{The Cancer Genome Atlas}
and \sigla{TARGET}{Therapeutically Applicable Research to Generate Effective Treatments},
\sigla{ENCODE}{The Encyclopedia of DNA Elements}, and
\sigla{ROADMAP}{The NIH Roadmap Epigenomics Mapping Consortium}.
For a better transparency, all tools will be open source and for their
scalability and interoperability they will be published in the Bioconductor project,
an environment that provides a broad range of powerful statistical and graphical methods
for the analysis of genomic data.
Furthermore, We also aim to investigate the intergenic epigenomic changes
associated with distinct biological and clinical subgroups of gliomas first
discovered by our laboratory \cite{ceccarelli2016molecular}. Specifically,
using the tools developed, we will integrate DNA methylation, gene expression,
mutation and copy number data as well as important epigenomic marks defined by
histone modifications in normal samples in order to identify candidate regulatory
elements associated with glioma progression.


\subsection{Specific aims}
\begin{enumerate}
    \item Download and process transcription factor (TF) ChIP-seq data for each cancer cell and tissue type through the ENCODE dataset;
    \item Download and process DNA methylation data for both cancer and non-tumor control cases through the TCGA consortium via HM450K platform;
    \item Identify statistically \sigla{DMR}{Differentially methylated regions} at the single CpG resolution;
    \item Determine statistically enriched proximal transcription factor binding sites (TFBSs) to altered DNA-methylated regions at the level of individual DNA/protein site interaction;
    \item Within known DMRs, classify and identify statistically known and novel DNA binding motifs;
    \item Download and process RNA-seq data from both cancer and non-tumor control cases;
    \item Use standard data structure to organize the data and the metadata;
    \item Correlate the DNA methylation status of \sigla{TFBSs}{Transcription factor binding sites} with target gene RNA-seq expression in order to determine regulatory networks that might alter the pan-cancer genome;
    \item Use learning machine algorithms for classifying an independent set of gliomas based on newly identified regulatory networks as related to pan-cancer deregulation;
    \item Develop tools to automate the previous steps;
    \item Use those tools to investigate the intergenic epigenomic changes associated with distinct biological and clinical subgroups of gliomas g-cimp-low and g-cimp-high discovered by our laboratory and collaborators;
    \item Compare the automated results with ones found manually in order to validate the package capacity in providing searching, retrieving and downstream biological analysis to discover pan-cancer epigenomic signatures able to redefine subgroups of gliomas;
    \item Submit those set of tools to be freely available in the open-source Bioconductor environment (available at \burl{http://www.bioconductor.org}).
\end{enumerate}

\section{Motivations}


Unravelling the genomic, epigenomic, and proteomic features using high-throughput methodologies is a central question for understanding regulatory gene networks in cancer. In this line of evidence, thousands of tumor and normal samples have been massively sequenced and a large amount of data are publicly deposited by the three main international consortia: TCGA, ENCODE, and Roadmap. However, a major challenge is the fact that the biological information necessary for a complete gene regulatory analysis is spread over different databases that store data in different formats. Moreover, to date there is a lack of  computational tools and methods that can integrate and interpret such information. Consequently, the process of analysis is performed manually by the end user, who must access all databases, select and process the data necessary to the project, and integrate that data using multiple downstream analysis tools to extract and interpret the relevant biological information.
To overcome these limitations, here we propose to implement tools for searching genomic and epigenomic data acquired from several biological databased, and to provide key scientific analysis steps and methodology, thereby allowing other researchers to apply  strategies for in-depth bioinformatics analysis. Those tools will be submitted to Bioconductor, and then, our expectation is that researchers may integrate all relevant data from the most important international consortia in the genomics field with their own experiments. Providing our package through Bioconductor, enables us to access a broader scientific community of advanced informatics users and developers worldwide, who can test the package with their own microarray and next-generation sequencing data, submit bug reports, criticize the methodology, provide new contributions, and ensure the quality of our package in terms of code and documentation.  In addition, storing the outputs inside an open-source software like R, allows one to utilize the many available statistical and analytical packages commonly used by researchers (\citealp{creditcode}). Then, we expect that these tools will provide insights and novel discoveries into unanticipated regulatory circuits in complex disease and normal developmental biology, which will be verified through the molecular analysis between the newly identified groups G-CIMP-low and G-CIMP-high.



\section{Organization of the Remainder of the Document}


This paper is organized into chapters as follows: In Chapter 2,
fundamental concepts required to the development of this project are introduced.
We review the concepts of cancer and their epigentic and genetic alterations,
followed by the description of the biological data generated through experiments and the data structure used for its storage,
finally we reviewed some of the data analysis methods used in the genetics field.
Chapters 3, 4, and 5 highlights the results of this work. Specifically, in Chapter 3,
we detail the methods and computational tools developed, while in Chapter 4 we
show their application in the real world. Enclosing this thesis, in Chapter 5,
we draw the main conclusions of this work, as well as the scientific contributions
derived from this project, possible future works and the main papers
that have been published during the Doctorate period.
